webrender/build_gfx.rs

```diff
@@ -341,20 +331,11 @@ fn extend_sampler_definition(
 }
 
 fn replace_non_sampler_uniforms(new_data: &mut String) {
-    new_data.push_str(
-        "\tlayout(push_constant) uniform PushConsts {\n\
-         \t\tmat4 uTransform;       // Orthographic projection\n\
-         \t\t// A generic uniform that shaders can optionally use to configure\n\
-         \t\t// an operation mode for this batch.\n\
-         \t\tint uMode;\n\
-         \t} pushConstants;\n",
-    );
     new_data.push_str(&format!(
         "\tlayout(set = {}, binding = 0) uniform Locals {{\n\
          \t\tuniform mat4 uTransform;       // Orthographic projection\n\
```

Comment:
why are we still having the transform here?

Reply:
I think we have a misunderstanding here, in #345 . you mentionedWebRender has it associated with the target, so we can safely move it into a constant buffer within the per-target descriptor set.

Reply:
I think we have a misunderstanding here, looks like I didn't payed attention what you mentioned in #345
> WebRender has it associated with the target, so we can safely move it into a constant buffer within the per-target descriptor set.

Unfortunately we don't have descriptor sets on a per target basis, we have the following groups: global (for each rendering loop), per WR pass and per draw. (The per-target variant would fit somewhere between second and third one.) That's the reason why I kept the uniform block around. If you think that getting rid of the uniform block and introducing a new descriptor binding with a new set index which points to a buffer could benefit us, I can make those changes as well.

Reply:
I think we can safely just adopt the "per pass" descriptor set and increase its frequency just a bit to bind whenever the projection happens. This is still relatively rare, i.e. dozens per frame, not thousands.

Reply:
Yes that's sounds totally reasonable. Also I will add a more descriptive name to it.